### DISCUSSION TITLE
How to multiply 2 number box values and show result on the 3rd number box in real-time?

### DISCUSSION THREAD

**Discussion by MikeYates** [Budibase/budibase#11649] [Help] [answered]
I have 2 number boxes, which I want to multiply the 'Amount' with 'Quantity' and generate result in realtime to a number box called 'Subtotal'

Amount x Quantity = Subtotal

I played around with update state and Condition but I cannot get the hang of it

**Comment by melohagan** [comment, marked as answer]
Hey @MikeYates 

You need to use the **Update Field Value** action in the *On Change* event of the amount and quantity fields. 

Docs: [On change event](https://docs.budibase.com/docs/form-bindings#on-change-event)

Because that involves multiple managing multiple actions, you may instead prefer to display the total in a paragraph component (or even an input within an Embed component).

Then you can simply bind the value: `{{ multiply Form.Fields.Amount Form.Fields.Quantity }}`

**Reply by dgagliardi** [reply]
Hi, I have a details side panel with 3 form fields, type number. I cannot get the value of the 3rd field to update (and show the update) in the field using the method above. In fact, I can't get the field to update at all, regardless of the formula. I even tried {{ multiply 5 2}} just to see? Second question, does it matter if the 3rd form field is disabled?

**Reply by melohagan** [reply]
Hey @dgagliardi Is this in the builder or the previewed app? 

Could you share some screenshots of your component tree, bindings etc please?

**Reply by dgagliardi** [reply]
Hi!
1. I do not see fields update in the previewed app
2. I have a MySQL DB table that I generated the various CRUD views.
3. I'm trying to update the 3rd form field from the values of the 2 other form fields in the details form block

Trying to get the form field totalPoints = numRounds * pointsPerRound

Thanks for the help!

<img width="1280" alt="image" src="https://github.com/Budibase/budibase/assets/709467/5bab7a61-6db4-455c-a2f1-0743b79971ea">


<img width="1280" alt="image" src="https://github.com/Budibase/budibase/assets/709467/9363002b-a51a-4d04-91ff-9e7b1ba09f7e">


<img width="1275" alt="image" src="https://github.com/Budibase/budibase/assets/709467/323b16c2-6a67-4725-a529-887c59ab2249">


<img width="1279" alt="image" src="https://github.com/Budibase/budibase/assets/709467/6b7a4a0d-2b27-4be7-8481-19a5a19473d9">

**Reply by dgagliardi** [reply]
@melohagan Hi!. Any thoughts? Thanks

**Reply by andz-bb** [reply]
hey @dgagliardi 

I have logged a bug for this as you aren't able to properly select the `Details_form_block` form when running the Update Field Value action. I have logged the issue internally and it will be picked up by another engineer soon.

In the meantime, the best way to work around this is to eject your Details Form Block. This will mean that is broken down to a parent Form component and the relevant field components. Once you do this, you should rename the Form component to something like _Details Form_. Now you can select the numRounds field and add an on change Update Field Value action. You'll be able to properly select the Details Form from the list as well as the totalPoints field you want to update.

<img width="878" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/52a51db7-698a-4104-9c98-e50f6c0ee6d3">


The handlebars binding for the updated totalPoints field value when updating the numRounds field should be:

```hbs
{{ multiply Field Value Details Form.Fields.pointsPerRound }}
```

The Field Value binding is the newly changed value for the field you're changing. With that in mind, when adding an on change action for the pointsPerRound field, the binding should be:

```hbs
{{ multiply Field Value Details Form.Fields.numRounds }}
```

Here's the final result - hope this helps 😊

![multiply fields](https://github.com/Budibase/budibase/assets/110921612/aa321e27-fc14-4e0d-9790-ecec112df4cd)

**Reply by dgagliardi** [reply]
Thanks. I actually ejected the block this morning to see if I could get it to work. Once I renamed the various forms from "Form" to something more specific, I was able to get the calculation to work but thanks for the validation!
